streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -144,6 +144,13 @@ class StreamingContext private[streaming] (
     }
   }
 
+  if (sc.conf.contains("spark.cores.max")) {
+    val totalCores = sc.conf.getInt("spark.cores.max", 1)
```

Comment:
@jiangxb1987 "spark.cores.max" is per application configuration to limit the numbers of cores can be requested for this application, it is not a per executor limitation. 
 
> The config spark.cores.max is used to limit the max number of cores that a single executor can require

So still if we have 2 receivers in one streaming application, the minimum number should > 2, checking "1" here is still not feasible.

Since receiver number can only be gotten in run-time, checking configuration will not be worked as expected.